src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java

```diff
@@ -30,39 +30,36 @@
 import java.util.stream.Collectors;
 
 /**
- * AlignmentService provides a way to align the active sites and backbones of proteins
+ * AlignmentService provides a way to align the active sites of proteins
  */
 @Service
 @Slf4j
 public class AlignmentService {
 
-    @Autowired
-    private ProteinService proteinService;
-
     @Autowired
     private MotifService motifService;
 
     /**
      * Executes the ActiveSiteAlignmentRequest on protein active sites.
      *
      * @param alignmentRequest The alignment request JSON mapped to an object
-     * @return AlignmentResponse which contains all alignments and their relevant data
+     * @return ActiveSiteAlignmentResponse which contains all alignments and their relevant data
      */
-    public AlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest alignmentRequest) {
-        return alignActiveSites(
-                proteinService.queryPdb(alignmentRequest.getPdbIds()),
-                alignmentRequest.getEcNumber()
-        );
-    }
+    public ActiveSiteAlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest alignmentRequest) {
+        List<Structure> sourceStructures = alignmentRequest.getPdbIdsAsStructures();
+        List<Motif> customMotifs = alignmentRequest.getCustomMotifs();
```

Comment:
Should we be paging these too?

Reply:
We shouldn't paginate custom motifs, but I can see an argument for paginating source structures. It's out of scope for this project iteration and would only be an optimization on enormous queries (thousands of PDB Ids)

Reply:
Sounds good!